### PR TITLE
Be explicit about what gets logged

### PR DESCRIPTION
**Pull request by MarcoPolo**
@keybase/react-hackers 

Explicitly pick what gets logged. Default to no payload data.

This is a simple pass to remove a bunch of logging. In a future PR I'll add more things to the whitelist (if you think of any, feel free to add here).

I think this is also a great use case for lenses since it should make describing paths much simpler and typesafe

### CODE REVIEW

**Comment by MarcoPolo** [comment]
This would replace https://github.com/keybase/client/pull/5047

**Comment by chrisnojima** [comment]
To clarify this a bit this came out of a conversation Marco and I had about the chat logging bug

**Comment by keybase-ci-visdiff** [comment]
The commits 3360206c59b9...2f4ebc5bcc3c introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Chat_Input-Emoji_Open** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1481679957940-3360206c59b9-2f4ebc5bcc3c-linux/Chat_Input-Emoji_Open-3360206c59b9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1481679957940-3360206c59b9-2f4ebc5bcc3c-linux/Chat_Input-Emoji_Open-2f4ebc5bcc3c.png)  
   Chat_Input-Emoji_Open comparison
</summary>

**Comment by keybase-ci-visdiff** [comment]
The commits 3360206c59b9...2f4ebc5bcc3c introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Chat_Input-Emoji_Open** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1481732033858-3360206c59b9-2f4ebc5bcc3c-linux/Chat_Input-Emoji_Open-3360206c59b9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1481732033858-3360206c59b9-2f4ebc5bcc3c-linux/Chat_Input-Emoji_Open-2f4ebc5bcc3c.png)  
   Chat_Input-Emoji_Open comparison
</summary>

File: shared/constants/reducer.js [hunk 1]
```diff
@@ -37,6 +37,27 @@ export type TypedState = {
   unlockFolders: UnlockFoldersState,
 }
 
+export type StateLogTransformer = (state: TypedState) => Object
+
 // TODO swap State with TypedState when TypedState includes everything we care about
 export type State = {[key: string]: any}
 export const stateKey = 'reducer:stateKey'
+
+// TODO expand this
```
Comment (chromakode) [inline shared/constants/reducer.js]: :sheep: expand this how?

File: shared/constants/types/flux.js [hunk 2]
```diff
@@ -20,3 +24,41 @@ export type Dispatch = (action: AsyncAction | Action) => ?Promise<*>
 
 export type TypedAsyncAction<A> = (dispatch: TypedDispatch<A>, getState: GetState) => ?Promise<*>
 export type TypedDispatch<-A> = (action: TypedAsyncAction<A> | A) => ?Promise<*>
+
+function _deepKeysOnly (m: Object) {
```
Comment (chromakode) [inline shared/constants/types/flux.js]: :sheep: These would be great candidates to unit test!

File: shared/constants/types/flux.js [hunk 3]
```diff
@@ -20,3 +24,41 @@ export type Dispatch = (action: AsyncAction | Action) => ?Promise<*>
...
+    if (typeof m[k] === 'object') {
+      acc[k] = _deepKeysOnly(m[k])
+    } else {
+      acc[k] = null
+    }
+    return acc
+  }, {})
+}
+
+function _shallowKeysOnly (m: Object) {
```
Comment (chromakode) [inline shared/constants/types/flux.js]: What is the expected output of this function? Looks like it'll just return an object with keys all set to null. Is that intended?

**Comment by keybase-ci-visdiff** [comment]
The commits 3360206c59b9...8613af7f9f33 introduce visual changes on linux. <details><summary>:mag_right: 1 changed</summary>

 * Changed: **Chat_Input-Emoji_Open** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1481844383524-3360206c59b9-8613af7f9f33-linux/Chat_Input-Emoji_Open-3360206c59b9.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1481844383524-3360206c59b9-8613af7f9f33-linux/Chat_Input-Emoji_Open-8613af7f9f33.png)  
   Chat_Input-Emoji_Open comparison
</summary>

**Pull request merged**